kythe/java/com/google/devtools/kythe/platform/shared/filesystem/ByteBufferByteChannel.java

```diff
@@ -0,0 +1,89 @@
+/*
+ * Copyright 2019 The Kythe Authors. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.devtools.kythe.platform.shared.filesystem;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ClosedChannelException;
+import java.nio.channels.NonWritableChannelException;
+import java.nio.channels.SeekableByteChannel;
+
+/** Read-only {@link SeekableByteChannel} backed by byte[] data. */
+class ByteBufferByteChannel implements SeekableByteChannel {
+  private byte[] buffer;
+  private long position;
```

Comment:
position should be an `int`

Reply:
It shouldn't. The API is specified in terms of long and the mixed math between int and long is isolated to the `read` function.